### PR TITLE
Fix handling of segments contained only of None values in sort merge

### PR DESCRIPTION
**Pull request by vasil-pashov**
#### Reference Issues/PRs
Monday: 9712540721

#### What does this implement or fix?
Sort and finalizes uses stream::Aggregator in combination with RowBuilder to combine the segments. The aggregator has row slicing policy controlled by a ConfigMap variable Merge.SegmentSize when the row builder accumulates more than Merge.SegmentSize rows (or the input is exhausted) it calls commit and writes the segment. The logic in do_merge skips `np.nan` and `None` values for string columns and does not add them to the corresponding column. It can happen so that a committed segment consist only of `np.nan` and `None` values. We don't handle this case properly in the encoding logic and throw an exception because the size of the column is 0 bytes.

#### Any other comments?
See Monday issues: 9712629170 and 9712849046
#### Checklist

<details>
  <summary>
   Checklist for code changes...
  </summary>
 
 - [ ] Have you updated the relevant docstrings, documentation and copyright notice?
 - [ ] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [ ] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [ ] Are API changes highlighted in the PR description?
 - [ ] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

### CODE REVIEW

File: cpp/arcticdb/stream/merge.hpp [hunk 1]
```diff
@@ -109,6 +110,21 @@ void do_merge(
                                 }
                             });
                         }
+                    } else {
+                        if constexpr(is_sequence_type(row_field_descriptor_tag.data_type())) {
```
Comment (vasil-pashov) [inline cpp/arcticdb/stream/merge.hpp:114]: This is the fix

File: python/tests/unit/arcticdb/version_store/test_sort_merge.py [hunk 2]
```diff
@@ -947,4 +947,12 @@ def test_appending_to_empty_with_differing_columns_fails(self, lmdb_library, to_
...
         lib.write(symbol, to_append, staged=True)
         with pytest.raises(SchemaException, match="wrong_col"):
-            lib.sort_and_finalize_staged_data(symbol, mode=StagedDataFinalizeMethod.APPEND)
\ No newline at end of file
+            lib.sort_and_finalize_staged_data(symbol, mode=StagedDataFinalizeMethod.APPEND)
+
+def test_staged_segment_has_only_none(lmdb_library):
+    symbol = "symbol"
+    lib = lmdb_library
+    df = pd.DataFrame({"a": [None, None, None]}, index=pd.DatetimeIndex([pd.Timestamp(1), pd.Timestamp(2), pd.Timestamp(3)]))
```
Comment (poodlewars) [inline python/tests/unit/arcticdb/version_store/test_sort_merge.py]: Also a test with, say, [None, "str", None]?
And similar tests where we have multiple row and column slices?

Comment (IvoDD) [reply python/tests/unit/arcticdb/version_store/test_sort_merge.py]: +1, maybe also a test with multiple staged segments some of which are empty and some not

**Review by alexowens90 — APPROVED** [review]

**Review by IvoDD — APPROVED** [review]
